crates/common/certificate/src/lib.rs

```diff
@@ -150,39 +150,39 @@ impl KeyCertPair {
-                hint: "The server certificate is not valid yet, the time it is being validated for is earlier than the certificate's `notBefore` time.".into(),
-                msg: msg.to_string(),
-            }),
-
-            msg if msg.contains("EndEntityUsedAsCa") => Some(CertificateError::CertificateValidationFailure {
-                hint: "An end-entity certificate is used as a server CA certificate. Make sure that the certificate used is signed by a correct CA certificate.".into(),
-                msg: msg.to_string(),
-            }),
-
-            msg if msg.contains("InvalidCertValidity") => Some(CertificateError::CertificateValidationFailure {
-                hint: "The server certificate validity period (`notBefore`, `notAfter`) is invalid, maybe the `notAfter` time is earlier than the `notBefore` time.".into(),
-                msg: msg.to_string(),
-            }),
-
-            _ => Some(CertificateError::CertificateValidationFailure {
-                hint: "Server certificate validation error.".into(),
-                msg: inner.to_string(),
-            }),
-        }
+                    rustls::CertificateError::Expired => Some(CertificateError::CertificateValidationFailure {
+                        hint: "The server certificate has expired, the time it is being validated for is later than the certificate's `notAfter` time.".into(),
+                        msg: err.to_string()
+                    }),
+
+                    rustls::CertificateError::NotValidYet => Some(CertificateError::CertificateValidationFailure {
+                        hint: "The server certificate is not valid yet, the time it is being validated for is earlier than the certificate's `notBefore` time.".into(),
+                        msg: err.to_string(),
+                    }),
+
+        //             msg if msg.contains("CaUsedAsEndEntity") => Some(CertificateError::CertificateValidationFailure {
+        //                 hint: "A CA certificate is used as an end-entity server certificate. Make sure that the certificate used is an end-entity certificate signed by CA certificate.".into(),
+        //                 msg: msg.to_string(),
+        //             }),
+
+        //             msg if msg.contains("EndEntityUsedAsCa") => Some(CertificateError::CertificateValidationFailure {
+        //                 hint: "An end-entity certificate is used as a server CA certificate. Make sure that the certificate used is signed by a correct CA certificate.".into(),
+        //                 msg: msg.to_string(),
+        //             }),
+
+        //             msg if msg.contains("InvalidCertValidity") => Some(CertificateError::CertificateValidationFailure {
+        //                 hint: "The server certificate validity period (`notBefore`, `notAfter`) is invalid, maybe the `notAfter` time is earlier than the `notBefore` time.".into(),
+        //                 msg: msg.to_string(),
+        //             }),
```

Comment:
These cases can simply be removed as there is no obvious match with actual [rustls::CertificateError](https://docs.rs/rustls/latest/rustls/enum.CertificateError.html).